### PR TITLE
refactor: refactor data request handler monitoring

### PR DESCRIPTION
**Pull request by samsja**
# Context

The behavior of the `receiving_request_seconds` is not perfect atm since it record the time for failed request as well.

# What this pr do ?

- [x] fix the behavior of the metric `receiving_request_seconds`
- [x] refactor the RequestHandler

### CODE REVIEW

File: jina/serve/runtimes/gateway/request_handling.py [hunk 1]
```diff
@@ -90,25 +89,53 @@ def _update_start_request_metrics(self, request: 'Request'):
...
+    :param metrics_registry: optional metrics registry for prometheus used if we need to expose metrics from the executor or from the data request handler
+    :param runtime_name: optional runtime_name that will be registered during monitoring
+    """
+
+    def __init__(
+        self,
+        metrics_registry: Optional['CollectorRegistry'] = None,
+        runtime_name: Optional[str] = None,
+    ):
+        super(RequestHandler, self).__init__(metrics_registry, runtime_name)
```
Comment (JohannesMessner) [inline jina/serve/runtimes/gateway/request_handling.py:135]: can't we just do `super().__init__(..)`?

File: jina/serve/runtimes/gateway/request_handling.py [hunk 2]
```diff
@@ -22,9 +22,9 @@
     from jina.types.request import Request
 
 
-class RequestHandler:
+class MonitoringRequestMixin:
```
Comment (JohannesMessner) [inline jina/serve/runtimes/gateway/request_handling.py:25]: Should we have this in a separate file, together with other monitoring related stuff? To have it all in one place and not clutter the main business logic anywhere?

**Review by alaeddine-13 — DISMISSED** [review]
LGTM

File: jina/serve/runtimes/gateway/request_handling.py (hunk 2 above)
Comment (samsja) [reply jina/serve/runtimes/gateway/request_handling.py:25]: I don't think that it should go into other monitoring things are they are speadout everywhere in the codebase. Putting it out of the file is a good idea though

**Pull request merged**